packages/story-editor/src/components/panels/shared/media/backgroundAudioPanelContent.js

```diff
@@ -298,7 +305,13 @@ function BackgroundAudioPanelContent({
             title={resource.src.split('/').pop()}
             isExternal={!resource.id}
             options={options}
-            onRemove={() => updateBackgroundAudio(null)}
+            onRemove={() => {
+              if (playerRef.current) {
+                playerRef.current.pause();
+              }
+
+              updateBackgroundAudio(null);
+            }}
```

Comment:
Should this be wrapped in a `useCallback`?

Reply:
Makes sense will update